### PR TITLE
feat: Introduce vehicle model lifecycle

### PR DESCRIPTION
**Pull request by doosuu**
# Description

Introduces the vehicle model lifecycle

## Azure DevOps PBI/Task reference

AB#_[PBI/Task number]_

## Checklist

* [ ] Vehicle App can be started with dapr run and is connecting to vehicle data broker
* [ ] Vehicle App can process MQTT messages and call the seat service
* [ ] Vehicle App can be deployed to local K3D and is running
* [ ] Created/updated tests, if necessary. Code Coverage percentage on new code shall be >= 70%.
* [ ] Extended the documentation in Velocitas repo
* [ ] Extended the documentation in README.md

* [ ] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [ ] Devcontainer can be re-built successfully

* [ ] Release workflow is passing

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,6 +43,10 @@ jobs:
         with:
           credentials: https://user:${{ secrets.GITHUB_TOKEN }}@github.com/
```
Comment (BjoernAtBosch) [inline .github/workflows/ci.yml:44]: Not fully sure, but I think, the "de-vri-es" could be removed: It's for accessing private repos ...

File: app/Dockerfile [hunk 2]
```diff
@@ -35,7 +36,8 @@ RUN pip3 install --no-cache-dir pyinstaller \
     && pip3 install --no-cache-dir patchelf==0.17.0.0 \
     && pip3 install --no-cache-dir staticx \
     && pip3 install --no-cache-dir -r ./app/requirements.txt \
-    && pip3 install --no-cache-dir -r ./app/requirements-links.txt
+    && pip3 install --no-cache-dir -r ./app/requirements-links.txt \
+    && pip3 install --no-cache-dir /vehicle_model
```
Comment (BjoernAtBosch) [inline app/Dockerfile:40]: This I don't understand fully:
* This is for app runtime container, right?
* When/where is that step done for local dev / CI?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dennismeister93) [reply .github/workflows/ci.yml:44]: at least. we should give the option to access private repos, when you generate a repo from the template it's not always public

File: app/Dockerfile (hunk 2 above)
Comment (dennismeister93) [reply app/Dockerfile:40]: this is explicitly for running the app in a container, yes!
For local execution the app/sourcecode has access to the ./gen/vehicle_model

If you start the k3d runtime and don't install the model upfront you get the following:

`kubectl logs sampleapp-8489757bf-mgd6z
Defaulted container "sampleapp" out of: sampleapp, daprd
sdv/dapr/_servicier.py:55: DeprecationWarning: There is no current event loop
Traceback (most recent call last):
  File "main.py", line 28, in <module>
ModuleNotFoundError: No module named 'vehicle'
[9] Failed to execute script 'main' due to unhandled exception!`

Comment (BjoernAtBosch) [reply app/Dockerfile:40]: Ok, installation as python package is done in the devenv-devcontainer-setup in /vehicle-model-lifecycle/src/generate_model.py

**Review by BjoernAtBosch — APPROVED** [review]
LGTM

**Pull request merged**